### PR TITLE
Return finalized block.

### PR DESCRIPTION
**Pull request by dvc94ch**

### CODE REVIEW

File: rosetta-types/src/network_status_response.rs [hunk 1]
```diff
@@ -43,7 +40,7 @@ impl NetworkStatusResponse {
         current_block_identifier: crate::BlockIdentifier,
         current_block_timestamp: i64,
         genesis_block_identifier: crate::BlockIdentifier,
-        latest_finalized_block_identifier: Option<crate::BlockIdentifier>,
+        latest_finalized_block_identifier: crate::BlockIdentifier,
```
Comment (Lohann) [inline rosetta-types/src/network_status_response.rs:43]: Ok so this should be a mandatory field.

**Review by Lohann — APPROVED** [review]

File: rosetta-client/src/wallet.rs [hunk 2]
```diff
@@ -143,10 +143,10 @@ impl Wallet {
         &self.account
     }
 
-    /// Returns the current block identifier.
+    /// Returns the latest finalized block identifier.
     pub async fn status(&self) -> Result<BlockIdentifier> {
         let status = self.client.network_status(self.config.network()).await?;
-        Ok(status.current_block_identifier)
+        Ok(status.latest_finalized_block_identifier)
```
Comment (Lohann) [inline rosetta-client/src/wallet.rs:149]: hmm.. who use this method? seems it is being used only in the ethereum examples directory.

Comment (dvc94ch) [reply rosetta-client/src/wallet.rs:149]: the timechain uses this method

Comment (dvc94ch) [reply rosetta-client/src/wallet.rs:149]: the timechain uses three  functions, `create_wallet`, `status` and `eth_view_call`. we could probably delete 99% of our connector code :slightly_smiling_face:

**Pull request merged**